src/pheval/analyse/benchmarking_data.py

```diff
@@ -0,0 +1,34 @@
+from dataclasses import dataclass
+from pathlib import Path
+
+from pheval.analyse.rank_stats import RankStats
+
+
+@dataclass
+class AnalysisResults:
+    """Analysis results for a run."""
+
+    results_dir: Path
+    ranks: dict
+    rank_stats: RankStats
+
+
+@dataclass
+class TrackRunPrioritisation:
+    """Track prioritisation for a run."""
+
+    gene_prioritisation: AnalysisResults = None
+    variant_prioritisation: AnalysisResults = None
+    disease_prioritisation: AnalysisResults = None
+
+    def return_gene(self) -> AnalysisResults:
+        """Return gene prioritisation analysis results."""
+        return self.gene_prioritisation
+
+    def return_variant(self) -> AnalysisResults:
+        """Return variant prioritisation analysis results."""
+        return self.variant_prioritisation
+
+    def return_disease(self) -> AnalysisResults:
+        """Return disease prioritisation analysis results."""
+        return self.disease_prioritisation
```

Comment:
Is the `TrackRunPrioritisation` dataclass actually necessary anymore? You have three types of `BenchmarkPrioritisationOutputGenerator` (can't this just be named `BenchmarkOutputGenerator`?) which explicitly calls a single method on the `TrackRunPrioritisation` instance, which returns one of three instances of `AnalysisResults`. Calling the wrong trackRunPrioritisation method will return `None` and the method called is dependent on the type of `BenchmarkOutputGenerator`, so why not just provide the correct `AnalysisResults` instance directly to the `BenchmarkPrioritisationOutputGenerator`  and be done with it?